include/xrpl/protocol/STEitherAmount.h

```diff
@@ -0,0 +1,313 @@
+
+    bool
+    isMPT() const;
+
+    bool
+    isIssue() const;
+
+    STEitherAmount const&
+    value() const;
+
+    std::variant<STAmount, STMPTAmount> const&
+    getValue() const;
+
+    std::variant<STAmount, STMPTAmount>&
+    getValue();
+
+    AccountID const&
+    getIssuer() const;
+
+    bool
+    negative() const;
+
+    bool
+    native() const;
+
+    STEitherAmount
+    zeroed() const;
+
+    int
+    signum() const noexcept;
+
+    template <ValidAmountType T>
+    T const&
+    get() const;
+
+    template <ValidAmountType T>
+    T&
+    get();
+
+private:
+    STBase*
+    copy(std::size_t n, void* buf) const override;
+    STBase*
+    move(std::size_t n, void* buf) override;
+};
+
+template <ValidAmountType T>
+T const&
+STEitherAmount::get() const
+{
+    if (std::holds_alternative<T>(amount_))
+        return std::get<T>(amount_);
+    Throw<std::logic_error>("Invalid STEitherAmount conversion");
+}
+
+template <ValidAmountType T>
+T&
+STEitherAmount::get()
+{
+    if (std::holds_alternative<T>(amount_))
+        return std::get<T>(amount_);
+    Throw<std::logic_error>("Invalid STEitherAmount conversion");
+}
+
+template <ValidAmountType T>
+decltype(auto)
+get(auto&& amount)
+{
+    using TAmnt = std::decay_t<decltype(amount)>;
+    if constexpr (std::is_same_v<TAmnt, STEitherAmount>)
+    {
+        if constexpr (std::is_lvalue_reference_v<decltype(amount)>)
+            return amount.template get<T>();
+        else
+            return T{amount.template get<T>()};
+    }
+    else if constexpr (std::is_same_v<TAmnt, std::optional<STEitherAmount>>)
+    {
+        static std::optional<T> t;
+        if (amount.has_value())
+            return std::make_optional(amount->template get<T>());
+        return t;
+    }
+    else if constexpr (std::is_convertible_v<TAmnt, STEitherAmount>)
+    {
+        if constexpr (std::is_lvalue_reference_v<decltype(amount)>)
+            return amount.operator STEitherAmount().template get<T>();
+        else
+            return T{amount.operator STEitherAmount().template get<T>()};
+    }
+    else
+    {
+        bool const alwaysFalse = !std::is_same_v<T, T>;
+        static_assert(alwaysFalse, "Invalid STEitherAmount conversion");
+    }
+}
+
+STEitherAmount
+amountFromJson(SField const& name, Json::Value const& v);
+
+STAmount
+amountFromJson(SF_AMOUNT const& name, Json::Value const& v);
+
+bool
+amountFromJsonNoThrow(STEitherAmount& result, Json::Value const& jvSource);
+
+bool
+amountFromJsonNoThrow(STAmount& result, Json::Value const& jvSource);
+
+inline bool
+operator==(STEitherAmount const& lhs, STEitherAmount const& rhs)
+{
+    return std::visit(
+        [&]<typename T1, typename T2>(T1 const& a1, T2 const& a2) {
+            if constexpr (std::is_same_v<T1, T2>)
+                return a1 == a2;
+            else
+                return false;
+        },
+        lhs.getValue(),
+        rhs.getValue());
+    if (lhs.isIssue() == rhs.isIssue())
```

Comment:
is this code reachable?

Reply:
Thanks. It's not reachable. I was debating what version to use and forgot to remove it.